src/libasr/pass/intrinsic_functions.h

```diff
@@ -2075,11 +2075,17 @@ namespace Dim {
 namespace Sqrt {
 
     static ASR::expr_t *eval_Sqrt(Allocator &al, const Location &loc,
-            ASR::ttype_t* arg_type, Vec<ASR::expr_t*> &args, diag::Diagnostics& /*diag*/) {
+            ASR::ttype_t* arg_type, Vec<ASR::expr_t*> &args, diag::Diagnostics& diag) {
         ASRUtils::ASRBuilder b(al, loc);
         if (is_real(*arg_type)) {
             double val = ASR::down_cast<ASR::RealConstant_t>(expr_value(args[0]))->m_r;
-            return b.f(std::sqrt(val), arg_type);
+            if (val < 0.0) {
+            append_error(diag, "Argument of SQRT has a negative value", loc);
+            return nullptr;
+            }
+            else{
+                return b.f(std::sqrt(val), arg_type);
+            }
```

Comment:
The else here seems redundant.

```suggestion
            return b.f(std::sqrt(val), arg_type);
```

Reply:
@Shaikh-Ubaid I have added the error message for the negative argument. But still, the function is returning nan values along with the error message.

```
cat a.f90
program sqrt_neg
    print *, sqrt(-1.0)
end program

lfortran a.f90
semantic error: Argument of SQRT has a negative value
 --> a.f90:2:14
  |
2 |     print *, sqrt(-1.0)
  |              ^^^^^^^^^^ 

Note: Please report unclear, confusing or incorrect messages as bugs at
https://github.com/lfortran/lfortran/issues.
         -nan
```

Reply:
https://github.com/lfortran/lfortran/blob/f6d9f0f9504939cfa143e949811f01265c7ef654/src/libasr/intrinsic_func_registry_util_gen.py#L704-L705

It looks like we need to add the following check in the above location.

```console
if (diagnostics.has_error()) {
    return nullptr
}

Reply:
We might need to add the same check after [this](https://github.com/lfortran/lfortran/blob/f6d9f0f9504939cfa143e949811f01265c7ef654/src/libasr/intrinsic_func_registry_util_gen.py#L694) line as well.